src/cswinrt/strings/additions/Windows.Foundation/Windows.Foundation.cs

```diff
@@ -25,8 +25,12 @@ public static Task AsTask(this IAsyncAction source, CancellationToken cancellati
                 throw new ArgumentNullException(nameof(source));
             }
 
-            // TODO: Handle the scenario where the 'IAsyncAction' is actually a task (i.e. originated from native code
-            // but projected into an IAsyncAction)
+            if (source is ITaskAwareAsyncInfo asyncInfo && asyncInfo.Task is Task task)
+            {
+                return cancellationToken.CanBeCanceled ?

```

Comment:
Did a CI build to confirm, but WaitAsync is .NET 6+ it seems so probably need a if NET guard around this to not run it on our NET standard projection.